src/frontend/src/functions/auth.tsx

```diff
@@ -150,24 +161,35 @@ export function checkLoginState(
     }
   };
 
-  // Check the 'user_me' endpoint to see if the user is logged in
   if (isLoggedIn()) {
-    api
-      .get(apiUrl(ApiEndpoints.user_me))
-      .then((response) => {
-        if (response.status == 200) {
-          loginSuccess();
-        } else {
-          loginFailure();
-        }
-      })
-      .catch(() => {
-        loginFailure();
-      });
+    // Already logged in
+    loginSuccess();
+    return;
+  }
+
+  // Not yet logged in, but we might have a valid session cookie
+  // Attempt to login
+  let result: boolean = false;
+
+  await api
+    .get(apiUrl(ApiEndpoints.user_me))
+    .then((response: any) => {
+      if (response.status == 200) {
+        result = true;
+      }
+    })
+    .catch(() => {});
+
+  if (result) {
+    await fetchUserState();
+  }
```

Comment:
Question: Can we just await fetchUserState? Seems like we will have 2 requests to user_me this way

Reply:
@matmair good call, I think you are correct. I'll make the change.